### PR TITLE
AutoReflect support for toggling several different projectile types

### PR DESCRIPTION
**Pull request by Rtas-17**
Mostly just modified IsEntStickyBomb and places where it's referenced to fit other projectiles. Feel like a lot of this could probably be made way more efficient, but it works I suppose

Created in response to #862

### CODE REVIEW

File: src/hacks/AutoReflect.cpp [hunk 1]
```diff
@@ -39,6 +51,88 @@ bool IsEntStickyBomb(CachedEntity *ent)
...
+{
+    if (ent->m_iClassID() == CL_CLASS(CTFProjectile_BallOfFire) || ent->m_iClassID() == CL_CLASS(CTFProjectile_EnergyBall))
+        return true;
+    else
+        return false;
+}
+
+bool IsEntCleaver(CachedEntity *ent)
+{
+    if (ent->m_iClassID() == CL_CLASS(CTFProjectile_BallOfFire) || ent->m_iClassID() == CL_CLASS(CTFProjectile_Cleaver))
```
Comment (BenCat07) [inline src/hacks/AutoReflect.cpp]: Wait what why is the Dragons fury fireball a cleaver?

**Comment by TotallyNotElite** [comment]
@Rtas-17 what's the status of this pull request?

File: src/hacks/AutoReflect.cpp (hunk 1 above)
Comment (Rtas-17) [reply src/hacks/AutoReflect.cpp]: Fixed that, my bad

**Comment by Rtas-17** [comment]
Removed the "balls" option entirely, as no pointer seems to exist to the any of the CTFProjectile_Ball variants. Fixing something like that is far beyond my skill level, so here's a band-aid for the time being

**Comment by TotallyNotElite** [comment]
image
this floor is made out of floor

**Comment by Rtas-17** [comment]
Think I've done as asked, and have also set big projectiles to be reflected as default, as it can get quite tedious going through and enabling those (and there's a very high likelihood that if someone is enabling auto-reflect to begin with, they're gonna want at-least the big projectiles to be blasted).

**Comment by TotallyNotElite** [comment]
image

**Comment by TotallyNotElite** [comment]
Tested this a bit, should be fine.

**Pull request merged**